pkg/resourceinterpreter/default/thirdparty/thirdparty_test.go

```diff
@@ -0,0 +1,187 @@
+package thirdparty
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative"
+	"github.com/karmada-io/karmada/pkg/resourceinterpreter/customized/declarative/luavm"
+	"github.com/karmada-io/karmada/pkg/util/interpreter"
+)
+
+var rules interpreter.Rules = interpreter.AllResourceInterpreterCustomizationRules
+
+func checkScrip(script string) error {
+	ctx, cancel := context.WithTimeout(context.TODO(), time.Second)
+	defer cancel()
+	l, err := luavm.NewWithContext(ctx)
+	if err != nil {
+		return err
+	}
+	defer l.Close()
+	_, err = l.LoadString(script)
+	return err
+}
+
+func getObj(t *testing.T, path string) *unstructured.Unstructured {
+	if path == "" {
+		return nil
+	}
+	data, err := os.ReadFile(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	jsonData, err := yaml.ToJSON(data)
+	if err != nil {
+		t.Fatal(err)
+	}
+	obj := make(map[string]interface{})
+	err = json.Unmarshal(jsonData, &obj)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return &unstructured.Unstructured{Object: obj}
+}
+
+func getAggregatedStatusItems(t *testing.T, path string) []workv1alpha2.AggregatedStatusItem {
+	if path == "" {
+		return nil
+	}
+	data, err := os.ReadFile(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var statusItems []workv1alpha2.AggregatedStatusItem
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(data), 4096)
+	for {
+		statusItem := &workv1alpha2.AggregatedStatusItem{}
+		err = decoder.Decode(statusItem)
+		if err != nil {
+			break
+		}
+		statusItems = append(statusItems, *statusItem)
+	}
+	if err != io.EOF {
+		t.Fatal(err)
+	}
+
+	return statusItems
+}
+
+type TestStructure struct {
+	Tests []IndividualTest `yaml:"tests"`
+}
+
+type IndividualTest struct {
+	DesiredInputPath  string `yaml:"desiredInputPath,omitempty"`
+	ObservedInputPath string `yaml:"observedInputPath,omitempty"`
+	StatusInputPath   string `yaml:"statusInputPath,omitempty"`
+	DesiredReplica    int64  `yaml:"desiredReplicas,omitempty"`
+	Operation         string `yaml:"operation"`
+}
+
+func checkInterpretationRule(t *testing.T, path string, configs []*configv1alpha1.ResourceInterpreterCustomization) {
+	ipt := declarative.NewConfigurableInterpreter(nil)
+	ipt.LoadConfig(configs)
+
+	dir := filepath.Dir(path)
+	yamlBytes, err := os.ReadFile(dir + "/customizations_tests.yaml")
```

Comment:
If we consider about the ostype, it can be `yamlBytes, err := os.ReadFile(dir + separator + "customizations_tests.yaml")`.

Reply:
Done.